### DISCUSSION TITLE
OpenAPI and Swagger

### DISCUSSION THREAD

**Discussion by rafacouto** [rust-lang-ve/hilow#8] [Ideas]
Is [Swagger](https://github.com/swagger-api/swagger-codegen) a good idea to define, document and generated stubs to the Hilow API REST?

**Comment by LeoBorai** [comment]
I think is awesome to have swagger support for this API documentation.

Lets dig in on how to provide swagger support for this project. Feel free to open an issue for this.

PD: @rafacouto , let's keep our GitHub content in english please. I think its important to be able to communicate with people who use different languages

**Comment by morenol** [comment]
It looks like this is not ready for production, but in any case it looks fine: https://github.com/wafflespeanut/paperclip

**Comment by rafacouto** [comment]
I want to explain tha swagger is a tool to define the API in the OpenAPI format (basically a yaml with the endpoint definitions). It is agnostic of languages but there are tools like swagger-codegen to generate server and client stubs. It is our decision to use in that way to generated the Hilow Rust API stub, but it is always useful to anyone implementing clients to consume Hilow API. Moreover, a pretty API documentation is generated using [swagger-hub](https://swagger.io/tools/swaggerhub/).

**Comment by LeoBorai** [comment]
Following this and previous discussions, we want to generate a Swagger documentation from the Actix server automatically.
Based on the services configured on Actix, we want to expose an endpoint, `http://myserver.com/swagger` for instance, where the documentation for this server is rendered as a swagger document.

A perfect example is HttpBin which makes use of Flasgger:
https://httpbin.org

Fastify in TypeScript-land achieves something similar through a Fastify's plugin:

[Fastify Swagger](https://github.com/fastify/fastify-swagger)

@morenol suggested https://github.com/wafflespeanut/paperclip as a possible solution for this.

**Comment by morenol** [comment]
In the Django world, [drf-yasg](https://github.com/axnsan12/drf-yasg) also achieves that
